hack/generator/pkg/astmodel/simple_type_definer.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+// SimpleTypeDefiner is a TypeDefiner for simple cases (not structs or enums)
+type SimpleTypeDefiner struct {
+	name        *TypeName
+	description *string
+	theType     Type
+}
+
+func NewSimpleTypeDefiner(name *TypeName, theType Type) *SimpleTypeDefiner {
+	return &SimpleTypeDefiner{name: name, theType: theType}
+}
+
+// SimpleTypeDefiner is a TypeDefiner
+var _ TypeDefiner = (*SimpleTypeDefiner)(nil)
+
+func (std *SimpleTypeDefiner) Name() *TypeName {
+	return std.name
+}
+
+func (std *SimpleTypeDefiner) Type() Type {
+	return std.theType
+}
+
+func (std *SimpleTypeDefiner) WithDescription(desc *string) TypeDefiner {
+	result := *std
+	result.description = desc
+	return &result
+}
+
+func (std *SimpleTypeDefiner) AsDeclarations() []ast.Decl {
+	var docComments *ast.CommentGroup
+	if std.description != nil {
+		docComments = &ast.CommentGroup{
+			List: []*ast.Comment{
+				{
+					Text: "\n/*" + *std.description + "*/",
```

Comment:
Leading and trailing spaces need to be a part of the comment.

Reply:
At the moment this adversely affects CRD generation. Perhaps we can fix it in `controller-gen`.

Reply:
Fair enough.